Bencodex/Decoder.cs

```diff
@@ -279,7 +279,46 @@ private void Back()
             _offset--;
         }
 
-        private byte[] ReadDigits(bool takeMinusSign, byte delimiter)
+        // Reads the length portion for byte strings and unicode strings.
+        private int ReadLength()
+        {
+            const byte colon = 0x3a;    // ':'
+            const int asciiZero = 0x30; // '0'
+            int length = 0;
+
+            var b = ReadByte();
+
+            if (b is null)
+            {
+                throw new DecodingException(
+                    $"Expected digits, but the byte stream terminates at {_offset}.");
+            }
+
+            byte lastByte = b.Value;
+            while (lastByte != colon)
+            {
+#pragma warning disable SA1131
+                if (lastByte < 0x30 || 0x39 < lastByte) // not '0'-'9'
```

Comment:
```suggestion
                if (!(0x30 <= lastByte && lastByte <= 0x39)) // not '0'-'9'
```

it might be helpful for readers.

Reply:
The original looks better to me.😉

Reply:
Eh, could be just a preference I suppose.

Reply:
Right. but aside from preferences, I believe a latter (i.e., `!(a <= b && b <= c)`) is more similar to intention (i.e., `b` isn't between `a` and `b`) than the former (i.e., `a < b || c < b`).